src/platform/silabs/efr32/BLEChannelImpl.cpp

```diff
@@ -0,0 +1,447 @@
+                                     0xC0 } };
+
+uint8_t InitialValueRX[16] = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 };
+uint8_t InitialValueTX[16] = { 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF };
+
+CHIP_ERROR MapBLEError(int bleErr)
+{
+    switch (bleErr)
+    {
+    case SL_STATUS_OK:
+        return CHIP_NO_ERROR;
+    case SL_STATUS_BT_ATT_INVALID_ATT_LENGTH:
+        return CHIP_ERROR_INVALID_STRING_LENGTH;
+    case SL_STATUS_INVALID_PARAMETER:
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    case SL_STATUS_INVALID_STATE:
+        return CHIP_ERROR_INCORRECT_STATE;
+    case SL_STATUS_NOT_SUPPORTED:
+        return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+    default:
+        return CHIP_ERROR(ChipError::Range::kPlatform, bleErr + CHIP_DEVICE_CONFIG_SILABS_BLE_ERROR_MIN);
+    }
+}
+
+} // namespace
+
+CHIP_ERROR BLEChannelImpl::Init()
+{
+    uint16_t session;
+    sl_status_t ret = sl_bt_gattdb_new_session(&session);
+    VerifyOrReturnError(ret == SL_STATUS_OK, MapBLEError(ret));
+
+    // Add service
+    ret = sl_bt_gattdb_add_service(session, sl_bt_gattdb_primary_service,
+                                   0, // not advertised
+                                   sizeof(kSideServiceUUID), kSideServiceUUID, &mSideServiceHandle);
+    VerifyOrReturnError(ret == SL_STATUS_OK, MapBLEError(ret));
+
+    // Add RX characteristic
+    ret = sl_bt_gattdb_add_uuid128_characteristic(session, mSideServiceHandle,
+                                                  SL_BT_GATTDB_CHARACTERISTIC_READ | SL_BT_GATTDB_CHARACTERISTIC_WRITE,
+                                                  0, // No security
+                                                  0, // No flags
+                                                  kRxUUID, sl_bt_gattdb_variable_length_value,
+                                                  255, // Max length
+                                                  sizeof(InitialValueRX), InitialValueRX, &mSideRxCharHandle);
+    VerifyOrReturnError(ret == SL_STATUS_OK, MapBLEError(ret));
+
+    ret = sl_bt_gattdb_add_uuid128_characteristic(session, mSideServiceHandle,
+                                                  SL_BT_GATTDB_CHARACTERISTIC_READ | SL_BT_GATTDB_CHARACTERISTIC_WRITE |
+                                                      SL_BT_GATTDB_CHARACTERISTIC_WRITE_NO_RESPONSE |
+                                                      SL_BT_GATTDB_CHARACTERISTIC_INDICATE,
+                                                  0, // No security
+                                                  0, // No flags
+                                                  kTxUUID, sl_bt_gattdb_variable_length_value,
+                                                  255, // Max length
+                                                  sizeof(InitialValueTX), InitialValueTX, &mSideTxCharHandle);
+    VerifyOrReturnError(ret == SL_STATUS_OK, MapBLEError(ret));
```

Comment:
There are too many calls to `MapBLEError` in the following lines, 
it would be best to change it to : 
```
VerifyOrExit(ret == SL_STATUS_OK);


exit : 
return MapBLEError(ret);
```

Reply:
We are not putting exit in our code anymore.